src/pages/Interferometry/MachZehnderCanvas.js

```diff
@@ -1,111 +1,21 @@
 import React, { useRef } from "react";
 import PropTypes from "prop-types";
 import useAnimationFrame from "../../useAnimation";
-import DoubleRect from "./doubleRect";
+import MachZehnderEquipment from "./MachZehnderEquipment";
 
-const source = new DoubleRect(50, 450, 50, 50, 0, 0, 1);
-
-const splitter1 = new DoubleRect(50, 350, 50, 5, 135, 2, 1);
-const splitter2 = new DoubleRect(250, 150, 50, 5, 135, 2, 1);
-
-const mirror1 = new DoubleRect(50, 150, 50, 5, 135, 1, 2);
-const mirror2 = new DoubleRect(250, 350, 50, 5, -45, 1, 2);
-
-const sample = new DoubleRect(50, 250, 50, -30, 0, 1, 0);
-
-function MachZehnderCanvas({ size: { width, height }, refractionIndex }) {
+function MachZehnderCanvas({ size: { width, height } }) {
   const cvs = useRef(null);
-
+  const equipment = new MachZehnderEquipment(
```

Comment:
that's better, but I think it's a constant so we can move it to the outside of the function

Reply:
Moved it!